provider/devices/tizen.go

```diff
@@ -0,0 +1,353 @@
+	device.SetupMutex.Lock()
+	defer device.SetupMutex.Unlock()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	device.ProviderState = "preparing"
+	logger.ProviderLogger.LogInfo("tizen_device_setup", fmt.Sprintf("Running setup for Tizen device `%v`", device.UDID))
+
+	err := cli.KillDeviceAppiumProcess(device.UDID)
+	if err != nil {
+		logger.ProviderLogger.LogError("tizen_device_setup", fmt.Sprintf("Failed attempt to kill existing Appium processes for device `%s` - %v", device.UDID, err))
+		ResetLocalDevice(device, "Failed to kill existing Appium processes.")
+		return
+	}
+
+	appiumPort, err := providerutil.GetFreePort()
+	if err != nil {
+		logger.ProviderLogger.LogError("tizen_device_setup", fmt.Sprintf("Could not allocate free host port for Appium for device `%v` - %v", device.UDID, err))
+		ResetLocalDevice(device, "Failed to allocate free host port for Appium")
+		return
+	}
+	device.AppiumPort = appiumPort
+
+	err = getTizenTVInfo(device)
+	if err != nil {
+		logger.ProviderLogger.LogError("tizen_device_setup", fmt.Sprintf("Failed to get TV info for device `%v` - %v", device.UDID, err))
+		ResetLocalDevice(device, "Failed to retrieve TV information.")
+		return
+	}
+
+	device.OS = "tizen"
+
+	go startAppium(device, &wg)
+	go checkAppiumUp(device)
+
+	select {
+	case <-device.AppiumReadyChan:
+		logger.ProviderLogger.LogInfo("tizen_device_setup", fmt.Sprintf("Successfully started Appium for device `%v` on port %v", device.UDID, device.AppiumPort))
+		break
+	case <-time.After(30 * time.Second):
+		logger.ProviderLogger.LogError("tizen_device_setup", fmt.Sprintf("Did not successfully start Appium for device `%v` in 60 seconds", device.UDID))
+		ResetLocalDevice(device, "Appium did not start within the expected time.")
+		return
+	}
+
+	device.ProviderState = "live"
+	wg.Wait()
+}
+
+func getTizenTVHost(tvID string) (string, error) {
+	// Check if the hostWithPort is in the format HOST_IP:PORT
+	if matched, _ := regexp.MatchString(`^([0-9]{1,3}\.){3}[0-9]{1,3}:\d+$`, tvID); matched {
+		host := strings.Split(tvID, ":")[0]
+		return host, nil
+	} else {
+		return "", fmt.Errorf("invalid format for host: %s", tvID)
+	}
+}
+
+func getTizenTVInfo(device *models.Device) error {
+	tvHost, err := getTizenTVHost(device.UDID)
+	if err != nil {
+		return fmt.Errorf("failed to get TV host - %s", err)
+	}
+
+	url := fmt.Sprintf("http://%s:8001/api/v2/", tvHost)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("failed to get TV info - %s", err)
+	}
+	defer resp.Body.Close()
+
+	var tvInfo TizenTVInfo
+	if err := json.NewDecoder(resp.Body).Decode(&tvInfo); err != nil {
+		return fmt.Errorf("failed to decode TV info - %s", err)
+	}
+
+	// Atualizar informações do device
+	device.Name = tvInfo.Device.Name
+	device.HardwareModel = tvInfo.Device.ModelName
+	device.OSVersion = tvInfo.Version
+	device.IPAddress = tvInfo.Device.IP
+	device.DeviceAddress = device.UDID
+
+	// Extrair dimensões da resolução
```

Comment:
English please :D

Reply:
Done in [commit](https://github.com/shamanec/GADS/pull/208/commits/a5e6765f6feb35691adabd2e558534bd5af65339)